processing/src/main/java/org/apache/druid/frame/write/RowBasedFrameWriter.java

```diff
@@ -235,7 +236,8 @@ private boolean mustSort()
     return rowOrderMemory != null;
   }
 
-  private boolean writeData()
+  @VisibleForTesting
+  boolean writeData()
```

Comment:
I think we should revert this change, and look for a way to generate the exception using the class's public methods (https://github.com/apache/druid/pull/11848#discussion_r739022142), if possible. Since the code coverage is still not satisfied, I think we can remove the new test added (since that isn't testing the actual code in the ControllerImpl where everything is bubbled up to) and merge it as is, since it's a fairly low risk change

Reply:
Thanks for the comment and the reference. Using a public method now instead.